### PR TITLE
Server

### PR DESCRIPTION
**Pull request by lymisscarol**
Add SupServer code to new branch "server".

### CODE REVIEW

File: server/SupServer.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      ex.printStackTrace();
+    }
+    catch(Exception ex) {
+      System.out.println(ex.getMessage());
+    }
+  }
+
+  // send the message to the specific user. find the socket by searching userlist
+  public void tellSomeone(String message, String name) {
+    while (userList.containsKey(name)) {
```
Comment (stevejarvis) [inline server/SupServer.java]: won't this infinite loop?

File: server/SupServer.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
+import java.io.*;
+import java.net.*;
+import java.util.*;
+
+public class SupServer {
+  
+  // userList is a hashmap which is used to store the user name and its printwriter
+  HashMap<String, PrintWriter> userList = new HashMap<String, PrintWriter>();
```
Comment (stevejarvis) [inline server/SupServer.java]: We'll need this map to be thread safe and somewhere all threads can get to it. Could create a [Singleton](http://stackoverflow.com/questions/11165852/java-singleton-and-synchronization) "Contacts" class with a [ConcurrentHashMap](http://docs.oracle.com/javase/6/docs/api/java/util/concurrent/ConcurrentHashMap.html)?

File: server/SupServer.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  }
+  
+  public void start () {
+    try{
+      ServerSocket supServer = new ServerSocket(3000);
+      
+      // everytime receive a socket from a server, create a new thread. put the printwriter into userlist and set the name for client
+      while(true) {
+        Socket supClient = supServer.accept();
+        PrintWriter writer = new PrintWriter(supClient.getOutputStream());
```
Comment (stevejarvis) [inline server/SupServer.java]: This has to be a 2 way connection, we need both in and out, but this only saved the out. We could create another Client object that keeps track of this socket?

**Comment by stevejarvis** [comment]
Also, this file needs to be in the java src directory (see the README). It doesn't do anything just put under server/ directory.

File: server/SupServer.java (hunk 1 above)
Comment (lymisscarol) [inline server/SupServer.java]: No, the while loop just open the portal for the corner and keep listening for new client socket

**Comment by stevejarvis** [comment]
@lymisscarol I added a contacts class and test to it, and moved your code into the source tree. I think your design makes sense overall. I hope you don't mind my editing your pull request (sorry!), I'm just getting a little worried about our getting a base down to get going with. I'll feel good when we start passing messages. Let me know if this makes sense to you and if you think it'll work.

**Comment by lymisscarol** [comment]
Thanks Steve, I really appreciate all the comments you give to me.

**Comment by stevejarvis** [comment]
Of course, if you're happy I say merge the PR. Normally there would need to be tests, but this is at a point before we can really test, since there's no client and it's hard to unit test socket code.

**Pull request merged**